### PR TITLE
Update GitBook docs with warning about core-type scaffold tests failing

### LINKED ISSUE #192: Update GitBook docs for creating core controllers to warn that tests need to updated
The generated tests for core types won't work out of the box because most core types don't have a valid empty values.

Update the GitBook documentation to communicate this.

### PR DESCRIPTION
**Pull request by pwittrock**
Closes #192

### CODE REVIEW

File: docs/book/beyond_basics/controllers_for_core_resources.md [hunk 1]
```diff
@@ -4,6 +4,11 @@ This book is being actively developed.
...
 ## Controllers for Core Resources
 
+{% panel style="warning", title="Core Only Projects" %}
+If your project will *only* contain controllers for types defined outside your project,
+you must create the project with the `--controller-only` flag.
+{% endpanel %}
+
 {% method %}
 
 It it possible to create Controllers for Core resources, or for resources defined outside your project.
```
Comment (lichuqiang) [inline docs/book/beyond_basics/controllers_for_core_resources.md]: nit: It is

File: docs/book/beyond_basics/controllers_for_core_resources.md [hunk 2]
```diff
@@ -14,7 +19,10 @@ kubebuilder create controller --group apps --version v1beta2 --kind Deployment -
...
 {% endmethod %}
 
-{% panel style="warning", title="Core Only Projects" %}
-If your project will *only* contain controllers for types defined outside your project,
-you must create the project with the `--controller-only` flag.
+{% panel style="warning", title="Scaffold Tests May Not Pass" %}
+When creating controllers for core and existing types, it may be necessary to modify
+the scaffold tests before they pass.
+
+This is because the empty Objet may not be valid and have required fields not
```
Comment (lichuqiang) [inline docs/book/beyond_basics/controllers_for_core_resources.md]: object

Comment (pwittrock) [reply docs/book/beyond_basics/controllers_for_core_resources.md]: Done

File: docs/book/beyond_basics/controllers_for_core_resources.md (hunk 1 above)
Comment (pwittrock) [reply docs/book/beyond_basics/controllers_for_core_resources.md]: Done

**Comment by pwittrock** [comment]
PTAL

**Pull request merged**